### ISSUE TITLE
Nav offcanvas remove "Browse All" from inserter block picker

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47014]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1374652632, when picking blocks from the Navigation offcanvas using the inserter there is a `Browse All` button. However, when clicked this breaks focus away from the offcanvas and should be removed.

**Comment by draganescu** [comment]
Three things I noticed trying to work on this issue:

1. The browse all button only seems to show up if the navigation block is inserted in the post editor.
2. What about the blocks that don't fit? Rely only on search?
3. In case we opt to leave it the way things happen is:
 - insert nav block
 - go to right hand list view
 - click on browse all
 - go to left hand inserter
 - insert block
 - left hand inserter closes
 - focus is on the newly inserted block
Is this OK?

**Comment by draganescu** [comment]
I seems an ancient #23737 introduced a check for `setInserterIsOpened` a method that comes from settings and somehow it is unavailable in the site editor. So far nobody complained about the browse all missing in the site editor but it is missing ever since I think? Why? Because removing the check for `setInserterIsOpened` makes the browse all buttons show up.

Not sure how to proceed :)

**Comment by getdave** [comment]
@ntsekouras Has been working on inserter a lot recently I think. Maybe he would know...?

**Comment by ntsekouras** [comment]
Hm.. It seems it's been removed  [here](https://github.com/WordPress/gutenberg/pull/44770/files#diff-53f39da7a5f1f996f5359a2dd7b2747fb3d60a494523dd73dba946fab8e82592R196). @youknowriad do you remember if there was there a specific reason for it?

**Comment by youknowriad** [comment]
Sorry, not sure I understand what was removed in my PR?

**Comment by youknowriad** [comment]
Ok I see what you mean, it was an inadvertent change, I'll fix it.

**Comment by draganescu** [comment]
I don't think we should remove the browse all button. Instead we should figure out something that tackles the context switching better. The problem in @alexstine 's review was that once the block was selected from the inserter the focus was not sent back to the list view in the inspector, but it was sent to the canvas instead.

Removing the browse all button hides blocks that can be inserted but don't fit in the six slots the quick inserter uses. There is search but one has to know what to search for.

**Comment by getdave** [comment]
> The problem in @alexstine 's review was that once the block was selected from the inserter the focus was not sent back to the list view in the inspector, but it was sent to the canvas instead.

See https://github.com/WordPress/gutenberg/issues/47018

> Removing the browse all button hides blocks that can be inserted but don't fit in the six slots the quick inserter uses. There is search but one has to know what to search for.

Ok let's close this one with the context that we didn't tackle it because there's a necessity to have the ability to browser all the blocks.
